domain/src/main/java/no/difi/meldingsutveksling/domain/arkivmelding/ArkivmeldingFactory.java

```diff
@@ -122,6 +126,12 @@ public Arkivmelding from(PutMessageRequestWrapper putMessage) {
 
         sm.getBasisregistrering().add(jp);
         am.getMappe().add(sm);
+        am.setSystem(properties.getNoarkSystem().getType());
+        am.setMeldingId(putMessage.getConversationId());
+        am.setTidspunkt(mt.getNoarksak().getSaDato().length() > 0
+                ? DateTimeUtil.toXMLGregorianCalendar(mt.getNoarksak().getSaDato())
+                : DateTimeUtil.toXMLGregorianCalendar(LocalDateTime.now().toString()));
```

Comment:
Dette vil ende opp i NPE dersom saDato er null (som den kan være dersom det ikke er verdi der). Se hvordan mappingen lenger opp håndterer det.
Trolig også tryggere å bruke DateTimeUtil.toXMLGregorianCalendar(OffsetDateTime) i stedet for å konvertere og parse fra String.

Reply:
Ev. bare bruke new GregorianCalendar(TimeZone)...